library/aes.c

```diff
@@ -1065,7 +1072,10 @@ int mbedtls_aes_crypt_cbc( mbedtls_aes_context *ctx,
         }
     }
 
-    return( 0 );
+end:
+    if( ret != 0 )
+        mbedtls_platform_zeroize( out, orig_length );
```

Comment:
If we can count on `mbedtls_aes_crypt_ecb()` to zeroize `output` when it fails, I don't think we need to zeroize `out` again here, since it should already be zeroized.

Reply:
This is in case we can count on that.
What if someone replaces only the `mbedtls_internal_aes_encrypt()` and `mbedtls_internal_aes_decrypt()` functions? For example, by using the Arm crypto extensions, and the operations fails in the middle? Can we assure that it clears the `out` buffer?

Reply:
OK

This is also in a loop which I didn't noticed before, so it makes sense to zeroize here.